CMakeLists.txt

```diff
@@ -317,7 +317,10 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   ADD_C_COMPILER_FLAG_IF_AVAILABLE("-Wno-unused-but-set-variable" HAVE_WNO_UNUSED_BUT_SET_VARIABLE)
   ADD_CXX_COMPILER_FLAG_IF_AVAILABLE("-Wno-unused-but-set-variable" HAVE_WNO_UNUSED_BUT_SET_VARIABLE)
 
-  find_package(SSE)
+  if (NOT ENABLE_GENERIC)
```

Comment:
i see your point but maybe the option should be named differently, like 'AUTO_DETECT_ISA` or so, and put the option into the main CMakeLists.txt with default value set to `true`. Also add the check in the `CMAKE_C_COMPILER` section